common/selectors/selectAndDedupe.js

```diff
@@ -13,11 +10,15 @@ dedupe.defaults.createKey = site => (site ? site.cache_key || site.hostname || s
  * @param {Array} group.defaults - default values to use.
  * @returns {Array}
  */
-module.exports = function selectAndDedupe(group) {
+module.exports.selectAndDedupe = function selectAndDedupe(group) {
   return group.reduce((result, options, index, arr) => {
     let current;
     let sites = options.sites;
 
+    // NOTE: The Math.random() fallback is because there can be empty slots thanks to pinning
+    // with a limited history.
+    dedupe.defaults.createKey = site => (site ? site.cache_key || site.hostname || site.url : Math.random());
```

Comment:
My intuition would be that the right approach here might be to dedupe before inserting the blank items, since deduping could affect the positioning of the blank items. 

On a higher level, I would argue handling deduping between the new sections adds way too much complexity as they start growing in number and getting resorted in terms of order, etc.